config/exec_resource.go

```diff
@@ -190,12 +195,52 @@ func (e *execResource) enforceState(ctx context.Context) (inDesiredState bool, e
 	case -1:
 		return false, err
 	case 100:
-		return true, nil
+		out, err := execOutput(ctx, e.GetEnforce().GetOutputFilePath())
+		e.enforceOutput = out
+		return true, err
 	default:
 		return false, fmt.Errorf("unexpected return code from enforce: %d, stdout: %s, stderr: %s", code, stdout, stderr)
 	}
 }
 
+func execOutput(ctx context.Context, outputFilePath string) ([]byte, error) {
+	if outputFilePath == "" {
+		return nil, nil
+	}
+
+	clog.Debugf(ctx, "Reading %q for ExecResource output", outputFilePath)
+
+	f, err := os.Open(outputFilePath)
+	if err != nil {
+		return nil, fmt.Errorf("error opening OutputFilePath: %v", err)
+	}
+	defer f.Close()
+
+	// Make a byte slice with a capacity of 1 over maxSize (for simplicity).
+	output := make([]byte, 0, maxExecOutputSize+1)
+	// Read up to capactity.
+	n, err := f.Read(output[:cap(output)])
+	output = output[:len(output)+n]
```

Comment:
My Golang skills are lacking here, but are there any places where the code will copy the bytes instead of updating or passing references?

Also, because this can be large, is there any value in calling GC() after sending this to the service instead of waiting on the runtime?

Reply:
No, we are working with slices here so we are always passing by reference.
As far as force calling GC(), perhaps, but this really isn't that much data and the runtime should be pretty on top of this, I expect reading any amount of data a user would notice will trigger the GC to run since normal usage is 2-3mb.
Ill do some testing and what we might be able to do is modify the GC percentage for the agent to trigger GC earlier since we really don't expect memory usage to go up that much